### ISSUE TITLE
YOLO export to ONNX Suddenly Fails Without Error or Output File

### ISSUE THREAD

**Issue by HenryParkG** [ultralytics/ultralytics#21653]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

<img width="1021" height="120" alt="Image" src="https://github.com/user-attachments/assets/85806b0f-13bd-47a6-996e-01e9a14d536a" />

I encountered a problem converting the "best.pt" model generated from training results to onnx. Normally, the file would be created within 5 seconds and visible in the file explorer, but this time, the command simply exited after outputting the verbose message above. Naturally, no file was created. The torch, CUDA, and onnx versions remained the same, but export suddenly stopped working. I'd like to know why.

The command I ran was very basic: 'yolo export model=best.pt format=onnx'

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HenryParkG, thanks for opening this issue and for using Ultralytics 🚀! This is an automated response to help you get support quickly. An Ultralytics engineer will review and assist you here soon.

Please start with the [Docs](https://docs.ultralytics.com/) where you’ll find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that may address common export questions.

Since this appears to be a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) so we can debug? For exports that silently exit, the following details are especially helpful:
-Exact command used and full console output with warnings/errors
-ultralytics and dependency versions (Python, PyTorch, CUDA, ONNX, onnxruntime if used)
-OS/GPU info
-The model file origin (which YOLO version trained it, any custom YAML)
-If possible, a small model/checkpoint that reproduces the issue

Join the Ultralytics community if you’d like faster back-and-forth while we investigate:
-For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
-Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/)
-Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Please upgrade to the latest ultralytics and verify all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to ensure this isn’t already resolved in the newest release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

-Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To move forward, please share the MRE and details above. This will help us quickly reproduce the ONNX export issue and suggest next steps 🔍✨

**Comment by RizwanMunawar** [comment]
@HenryParkG, can you please upgrade the ultralytics package to the latest version with the command `pip install -U ultralytics` and try again? I hope it will resolve your issue. If not, please share the new error logs, and I will investigate. FYI, on myside everything is working fine for now. Thanks

**Comment by HenryParkG** [comment]
<img width="426" height="721" alt="Image" src="https://github.com/user-attachments/assets/96a5380d-5bb0-4bca-b6b0-30e59a534748" />


I've already tried the latest version. (Requirement already satisfied)
It would be nice to see an error log, but the command terminates without creating the file, ending with 'ONNX: starting export with onnx 1.17.0 opset 19...'.

Are there any possible causes? (Perhaps the file is corrupted?)
Thank you for your reply.

**Comment by RizwanMunawar** [comment]
@HenryParkG, ok, can you try exporting the pre-trained Ultralytics YOLO11 model with the command:

```bash
yolo export format=onnx model=yolo11n.pt
```

If that works, then probably the file is corrupted.

**My results**

<img width="1289" height="455" alt="Image" src="https://github.com/user-attachments/assets/70fcae59-97aa-4ec9-a37f-d3d2a7b2aa4c" />

**Comment by HenryParkG** [comment]
<img width="1070" height="113" alt="Image" src="https://github.com/user-attachments/assets/c508e297-84c1-4fff-adac-665e3d27bbca" />

I tried that command, but it still doesn't get to 'slimming'.
And tried it in various prompts such as cmd and powershell, but it was the same.

I tried another attempt. I uninstalled and reinstalled Ultralytics, and moved it from the D drive to the C drive, but the result was the same.

**Comment by RizwanMunawar** [comment]
Can you share the output logs of this command, `yolo checks`?

**Comment by HenryParkG** [comment]
<img width="939" height="591" alt="Image" src="https://github.com/user-attachments/assets/7e17a1d8-f387-44e5-9144-73317d94e92e" />

Here is the command result

**Comment by RizwanMunawar** [comment]
@HenryParkG I don't see any reason the onnx export is failing, I tried the same workflow on my machine and everything is working fine.

Maybe you can check this issue, https://github.com/ultralytics/ultralytics/issues/20169 or this one https://github.com/ultralytics/ultralytics/issues/20423

**Comment by HenryParkG** [comment]
'yolo export format=onnx model=yolo11n.pt device=0'

Referencing that issue, I ran the above test command to resolve the issue. I confirmed that it was applicable to my actual model file.
Previously, it worked fine even without the 'device=0' option. This is amazing.

Thank you.

**Comment by RizwanMunawar** [comment]
Glad to hear this @HenryParkG :)
